rpcs3/Emu/Cell/PPUInterpreter.cpp

```diff
@@ -4716,12 +4714,15 @@ bool ppu_interpreter::MTFSF(ppu_thread& ppu, ppu_opcode_t op)
 
 bool ppu_interpreter::FCMPU(ppu_thread& ppu, ppu_opcode_t op)
 {
-	const f64 a = ppu.fpr[op.fra];
-	const f64 b = ppu.fpr[op.frb];
-	ppu.fpscr.fg = a > b;
-	ppu.fpscr.fl = a < b;
-	ppu.fpscr.fe = a == b;
-	//ppu.fpscr.fu = a != a || b != b;
+	if (const f64 a = ppu.fpr[op.fra]; true)
```

Comment:
Syntax abuse, again. What's the point of having a scope here?

Reply:
Its to avoid assigning too many registers at a time for the compiler, keep in mind ppu_cr_set is inlined and assigns regs as well.
I'll reduce scope usage to only what benefits compiled code.

Reply:
That's an interesting claim. However I have doubts that even MSVC compiler is so bad it can't handle it automatically. I tried mostly newer versions of GCC and Clang, MSVC too (with /Ox parameter): https://godbolt.org/z/U8-FWF
Adding or removing the scope doesn't seem to affect generated code at all.

Reply:
I won't say the example copies RPCS3 exactly - I'd like to see actual difference if it exists.

Reply:
ok checked it and there is no difference.